src/topology/metric_space/metrizable.lean

```diff
@@ -0,0 +1,116 @@
+    real.dist_le_of_mem_Icc_01 (hf01 _ _) (hf01 _ _)⟩,
+  have hF : ∀ x UV, F x UV = f UV x := λ _ _, rfl,
+  refine ⟨F, embedding.mk' _ (λ x y hxy, _) (λ x, le_antisymm _ _)⟩,
+  { /- First we prove that `F` is injective. Indeed, if `F x = F y` and `x ≠ y`, then we can find
+    `(U, V) ∈ s` such that `x ∈ U` and `y ∉ V`, hence `F x UV = 0 ≠ ε UV = F y UV`. -/
+    refine not_not.1 (λ Hne, _), -- `by_contra Hne` timeouts
+    rcases hB.mem_nhds_iff.1 (is_open_ne.mem_nhds Hne) with ⟨V, hVB, hxV, hVy⟩,
+    rcases hB.exists_closure_subset (hB.mem_nhds hVB hxV) with ⟨U, hUB, hxU, hUV⟩,
+    set UV : ↥s := ⟨(U, V), ⟨hUB, hVB⟩, hUV⟩,
+    apply (ε01 UV).1.ne,
+    calc (0 : ℝ) = F x UV : (hf0 UV hxU).symm
+             ... = F y UV : by rw hxy
+             ... = ε UV   : hfε UV (λ h : y ∈ V, hVy h rfl) },
+  { /- Now we prove that each neighborhood `V` of `x : X` include a preimage of a neighborhood of
+    `F x` under `F`. Without loss of generality, `V` belongs to `B`. Choose `U ∈ B` such that
+    `x ∈ V` and `closure V ⊆ U`. Then the preimage of the `(ε (U, V))`-neighborhood of `F x`
+    is included by `V`. -/
+    refine ((nhds_basis_ball.comap _).le_basis_iff hB.nhds_has_basis).2 _,
+    rintro V ⟨hVB, hxV⟩,
+    rcases hB.exists_closure_subset (hB.mem_nhds hVB hxV) with ⟨U, hUB, hxU, hUV⟩,
+    set UV : ↥s := ⟨(U, V), ⟨hUB, hVB⟩, hUV⟩,
+    refine ⟨ε UV, (ε01 UV).1, λ y (hy : dist (F y) (F x) < ε UV), _⟩,
+    replace hy : dist (F y UV) (F x UV) < ε UV,
+      from (bounded_continuous_function.dist_coe_le_dist _).trans_lt hy,
+    contrapose! hy,
+    rw [hF, hF, hfε UV hy, hf0 UV hxU, pi.zero_apply, dist_zero_right],
+    exact le_abs_self _ },
+  { /- Finally, we prove that `F` is continuous. Given `δ > 0`, consider the set `T` of `(U, V) ∈ s`
+    such that `ε (U, V) ≥ δ`. Since `ε` tends to zero, `T` is finite. Since each `f` is continuous,
+    we can choose a neighborhood such that `dist (F y (U, V)) (F x (U, V)) ≤ δ` for any
+    `(U, V) ∈ T`. For `(U, V) ∉ T`, the same inequality is true because both `F y (U, V)` and
+    `F x (U, V)` belong to the interval `[0, ε (U, V)]`. -/
+    refine (nhds_basis_closed_ball.comap _).ge_iff.2 (λ δ δ0, _),
+    have h_fin : finite {UV : s | δ ≤ ε UV}, by simpa only [← not_lt] using hε (gt_mem_nhds δ0),
+    have : ∀ᶠ y in 𝓝 x, ∀ UV, δ ≤ ε UV → dist (F y UV) (F x UV) ≤ δ,
+    { refine (eventually_all_finite h_fin).2 (λ UV hUV, _),
+      exact (f UV).continuous.tendsto x (closed_ball_mem_nhds _ δ0) },
+    refine this.mono (λ y hy, (bounded_continuous_function.dist_le δ0.le).2 $ λ UV, _),
+    cases le_total δ (ε UV) with hle hle,
+    exacts [hy _ hle, (real.dist_le_of_mem_Icc (hf0ε _ _) (hf0ε _ _)).trans (by rwa sub_zero)] }
+end
+
+/-- A normal topological space with second countable topology `X` is metrizable: there exists a
+metric space structure that generates the same topology. -/
+lemma metrizable_of_normal_space_second_countable_topology :
```

Comment:
could you also register this as a def (i.e., define a metric space structure for which the topology is definitionally the original one)? I feel this will be easier to use when we need indeed to endow such a space with a metric space structure.

Reply:
Replaced the lemma with a def. We use classical logic anyway.